src/test/java/uk/gov/hmcts/reform/unspec/service/docmosis/cos/CertificateOfServiceGeneratorTest.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.hmcts.reform.unspec.service.docmosis.cos;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.jackson.JacksonAutoConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.unspec.model.CaseData;
+import uk.gov.hmcts.reform.unspec.model.docmosis.DocmosisData;
+import uk.gov.hmcts.reform.unspec.model.docmosis.DocmosisDocument;
+import uk.gov.hmcts.reform.unspec.model.documents.CaseDocument;
+import uk.gov.hmcts.reform.unspec.model.documents.Document;
+import uk.gov.hmcts.reform.unspec.model.documents.PDF;
+import uk.gov.hmcts.reform.unspec.service.docmosis.DocumentGeneratorService;
+import uk.gov.hmcts.reform.unspec.service.documentmanagement.DocumentManagementService;
+import uk.gov.hmcts.reform.unspec.utils.ResourceReader;
+
+import java.time.LocalDateTime;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.unspec.callback.CallbackHandlerFactoryTest.BEARER_TOKEN;
+import static uk.gov.hmcts.reform.unspec.model.documents.DocumentType.CERTIFICATE_OF_SERVICE;
+import static uk.gov.hmcts.reform.unspec.service.docmosis.DocmosisTemplates.N215;
+import static uk.gov.hmcts.reform.unspec.service.documentmanagement.DocumentManagementService.UNSPEC;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {
+    CertificateOfServiceGenerator.class,
+    JacksonAutoConfiguration.class
+})
+class CertificateOfServiceGeneratorTest {
+
+    public static final String REFERENCE_NUMBER = "000LR095";
+    private final byte[] bytes = {1, 2, 3, 4, 5, 6};
+    private final String fileName = format(N215.getDocumentTitle(), REFERENCE_NUMBER);
+
+    @MockBean
+    private DocumentManagementService documentManagementService;
+    @MockBean
+    private DocumentGeneratorService documentGeneratorService;
+
+    @Autowired
+    private CertificateOfServiceGenerator certificateOfServiceGenerator;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Test
+    @SneakyThrows
+    void shouldGenerateCertificateOfService() {
```

Comment:
should_when for test names

Reply:
do you see any value in extra tests for different service locations etc etc?

Reply:
Nope as those are covered in it's individual tests now.